src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java

```diff
@@ -1491,78 +1491,57 @@ protected final void setSkid(String skidValue){
-                return FormValidation.error("Must specify either IAM Role ARN *OR* AKID/SKID!");
+        private AWSDeviceFarm getDeviceFarmInstance(final String roleArn,
+                                                    final Secret akid,
+                                                    final Secret skid) {
+            AWSDeviceFarm adf;
+            if (roleArn == null || roleArn.isEmpty()) {
+                return new AWSDeviceFarm(new BasicAWSCredentials(Secret.toString(akid), Secret.toString(skid)));
+            } else {
+                return new AWSDeviceFarm(roleArn);
             }
-            return FormValidation.ok();
         }
 
         /**
-         * Validate the user account AKID.
-         *
-         * @param akid The AWS access key ID.
-         * @return Whether or not the form was ok.
+         * Validates if the credentials provided for the verification are valid or not.
+         * @param roleArn the role arn
+         * @param akid the access key
+         * @param skid the secret key
+         * @return result of the validation
          */
-        @SuppressWarnings("unused")
-        public FormValidation doCheckAkid(@QueryParameter String akid) {
-            String skid = Secret.toString(this.skid);
-            if ((roleArn == null || roleArn.isEmpty()) && (akid == null || akid.isEmpty())) {
-                return FormValidation.error("Required if IAM Role ARN isn't present!");
-            }
-            if (roleArn != null && !roleArn.isEmpty() && akid != null && !akid.isEmpty() && skid != null && !skid.isEmpty()) {
-                return FormValidation.error("Must specify either IAM Role ARN *OR* AKID/SKID!");
-            }
-            return FormValidation.ok();
-        }
+        public FormValidation doValidateCredentials(@QueryParameter("roleArn") final String roleArn,
+                                                    @QueryParameter("akid") final String akid,
+                                                    @QueryParameter("skid") final String skid) {
+            try {
 
-        /**
-         * Validate the user account SKID.
-         *
-         * @param skid The AWS secret key ID.
-         * @return Whether or not the form was ok.
-         */
-        @SuppressWarnings("unused")
-        public FormValidation doCheckSkid(@QueryParameter String skid) {
-            String akid = Secret.toString(this.akid);
-            if ((roleArn == null || roleArn.isEmpty()) && (skid == null || skid.isEmpty())) {
-                return FormValidation.error("Required if IAM Role ARN isn't present!");
-            }
-            if (roleArn != null && !roleArn.isEmpty() && akid != null && !akid.isEmpty() && skid != null && !skid.isEmpty()) {
-                return FormValidation.error("Must specify either IAM Role ARN *OR* AKID/SKID!");
+                if ((roleArn == null || roleArn.isEmpty()) && (akid == null || akid.isEmpty() || skid == null || skid.isEmpty())) {
```

Comment:
Improve the conditioning. Use null safe check if possible

Reply:
used stringutils.